### ISSUE TITLE
[TW-1610] "task <n> edit" brings up wrong task when gc=off

### ISSUE THREAD

**Issue by tbabej** [tbabej/testimportfinal#1571]
_Scott Mcdermott on 2015-05-04T01:42:14Z says:_ 

 I had some deleted tasks.  I set `gc=off`, and run it manually every so often.

I just did a `task list` and saw my task 2.  I then did `task 2 edit` to change the description.  To my surprise, it brought up an old task, with deleted status! Totally different task than the one `task list` shows.  The old task showed up in the editor with id=2 !!

Sure enough, when I ran `task rc.gc=on rc.verbose=nothing // list` this restored `task 2 edit` to open up the correct one (as shown by `task list`)!

This seems like a clear bug:

# it's editing the wrong task entirely, and
# it depends on gc having been run to edit the right one

**Comment by tbabej** [comment]
Migrated metadata:
```
Created: 2015-05-04T01:42:14Z
Modified: 2015-05-04T07:33:08Z```

**Comment by tbabej** [comment]
_Wim Schuermann on 2015-05-04T07:33:08Z says:_ 

 Damn, this is a bad one. Thank you for reporting it, Scott!

The changes I did to fix an earlier problem with invalid IDs in TW-1583 led to the behavior you noticed here when GC is switched off by the user. It also affects "mod". I'll have to repeat what I said above: Damn, this is a bad one.

Fixed in the 2.4.4 dev branch.
